src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -884,9 +931,13 @@ def delete_service_principal_credential(cmd, identifier, key_id, cert=False):
         key_id, identifier))
 
 
-def _resolve_service_principal(client, identifier):
+def _resolve_service_principal(client, identifier=None, appid=None):
```

Comment:
Can you please double check this is needed?  appid and app id uri are both service principal names which the existing query below should cover it. I did a quick check just now for reference, but let me know if you do need it.
```
(env) D:\sdk\azure-cli>az ad app show --id e96b494b-61a2-4a42-a33c-e11ab1f1a6b6
{
  <omit irrelevant fields>
  "appId": "e96b494b-61a2-4a42-a33c-e11ab1f1a6b6",
  ...
}
(env) D:\sdk\azure-cli>az ad sp show --id e96b494b-61a2-4a42-a33c-e11ab1f1a6b6
{
  "appDisplayName": "yugangw-op2",
  "appId": "e96b494b-61a2-4a42-a33c-e11ab1f1a6b6",
  "appOwnerTenantId": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a",
   "objectId": "e0a58f57-dcae-48ab-98b6-ee173a6097fa",
  "objectType": "ServicePrincipal",
   servicePrincipalNames": [
    "http://yugangw-op2",
    "e96b494b-61a2-4a42-a33c-e11ab1f1a6b6"
  ],
  "servicePrincipalType": "Application",
   <omit irrelevant fields>
}
```

Reply:
@yugangw-msft I can confirm that the appId can be the same as the objectId, but it is not always. In all of my test cases, like for example creating an app in the portal, the SP associated with the app will be autogenerated and needs to be searched for by the appId. The code is definitely needed.

Reply:
object id is irrelevant here which is different thing from the app id. I am talking about the app id of the application. The service principal provisioned in the tenant for the application should have the app id as part of the `servicePrincipalNames` array. If you see the appid is not in the `servicePrincipalNames` of the SP then your change is definitely needed. To get my comment clear, I am talking about the following array in the SP object
```bash
   servicePrincipalNames": [
    "http://yugangw-op2",
    "e96b494b-61a2-4a42-a33c-e11ab1f1a6b6"
  ],
```

Reply:
Not sure if I am missing something here, this is the flow:
1) I have the appId/clientId on my app
2) I need to get the objectId of the SP associated with my app
3) The app array does not contain the objectId of the SP
4) I cannot query the SP by the objectId as I do not have the objectId, I only have the appId
5) I query the SP by the appId which gives me the SP ObjectId

Reply:
The flow is right. My comment is on step 5 that you don't need to update `_resolve_service_principal` to take a new argument of `appid`, rather just pass the app id to the existing argument of `identifier` which should work.
The reason is the `_resolve_service_principal` queries for SP using `servicePrincipalNames`, and `appid` is service principal name, so the query should just work.

Reply:
@yugangw-msft , thank you, in my tests this 
>rather just pass the app id to the existing argument of identifier which should work
 was not working for me

Reply:
thanks. Is it easy for you to share a repro for me to try out?

Reply:
@yugangw-msft , you are right - my tests were bad, will remove appId from query and rebase